src/models/UploadsManagerStore.ts

```diff
@@ -0,0 +1,202 @@
+  contentId: types.identifier,
+  parentObject: ParentObject,
+  owner: types.string,
+  type: types.enumeration(['video', 'thumbnail', 'cover', 'avatar']),
+  lastStatus: types.maybe(types.enumeration(['completed', 'inProgress', 'error', 'reconnecting', 'reconnectionError'])),
+  liaisonJudgement: types.maybe(
+    types.enumeration<LiaisonJudgement>('LiaisonJudgement', Object.values(LiaisonJudgement))
+  ),
+  ipfsContentId: types.maybe(types.string),
+  // size in bytes
+  size: types.maybe(types.number),
+  dimensions: types.maybe(AssetDimensions),
+  imageCropData: types.maybe(ImageCropData),
+  metadata: types.maybe(types.string),
+  title: types.maybe(types.union(types.string, types.null)),
+  progress: types.optional(types.number, 0),
+})
+
+export const UploadsManagerStore = types
+  .model('UploadsManagerStore', {
+    uploadingAssetsState: types.array(AssetUpload),
+  })
+  .views((self) => ({
+    // get count() {
+    //   return self.x.length
+    // },
+  }))
+  .volatile((self) => {
+    const assetsFiles: AssetFile[] = []
+    const pendingNotificationsCounts = { uploading: 0, uploaded: 0 }
+
+    const displayUploadingNotification = debounce(() => {
+      getRoot<RootStoreIntance>(self).hooks.snackbar.displaySnackbar?.({
+        title:
+          pendingNotificationsCounts.uploading > 1
+            ? `${pendingNotificationsCounts.uploading} assets being uploaded`
+            : 'Asset being uploaded',
+        iconType: 'info',
+        timeout: UPLOADED_SNACKBAR_TIMEOUT,
+        actionText: 'See',
+        // TODO:
+        // onActionClick: () => navigate(absoluteRoutes.studio.uploads()),
+      })
+      pendingNotificationsCounts.uploading = 0
+    }, 700)
+    const displayUploadedNotification = debounce(() => {
+      getRoot<RootStoreIntance>(self).hooks.snackbar.displaySnackbar?.({
+        title:
+          pendingNotificationsCounts.uploaded > 1
+            ? `${pendingNotificationsCounts.uploaded} assets uploaded`
+            : 'Asset uploaded',
+        iconType: 'success',
+        timeout: UPLOADED_SNACKBAR_TIMEOUT,
+        actionText: 'See',
+        // TODO:
+        // onActionClick: () => navigate(absoluteRoutes.studio.uploads()),
+      })
+      pendingNotificationsCounts.uploaded = 0
+    }, 700)
+
+    return { assetsFiles, pendingNotificationsCounts, displayUploadingNotification, displayUploadedNotification }
+  })
+  .actions((self) => {
+    function addAsset(asset: SnapshotOrInstance<typeof AssetUpload>) {
+      self.uploadingAssetsState.push(asset)
+    }
+    function updateAsset(asset: SnapshotOrInstance<typeof AssetUpload>) {
+      const index = self.uploadingAssetsState.findIndex((a) => a.contentId === asset.contentId)
+      if (index >= 0) {
+        self.uploadingAssetsState[index] = { ...self.uploadingAssetsState[index], ...cast(asset) }
+      }
+    }
+    function removeAsset(asset: SnapshotOrInstance<typeof AssetUpload>) {
+      self.uploadingAssetsState.remove(cast(asset))
```

Comment:
How does this work? `remove` is some special function for removing an element from an array?

Reply:
Since this is an observable array it has some extra utils built on top remove is one of them and it removes the element from the array
https://mobx.js.org/api.html#collection-utilities-